### PR TITLE
caprice32 : init at unstable-2018-02-10

### PR DESCRIPTION
**Pull request by bignaux**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/emulators/caprice32/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "caprice32";
+  version = "20180210";
+  rev = "53de69543300f81af85df32cbd21bb5c68cab61e";
+  short_rev = "${builtins.substring 0 7 rev}";
+  name = "${pname}-${version}-${short_rev}";
+
+  src = fetchFromGitHub {
+    owner = "ColinPitrat";
+    repo  = "caprice32";
+    rev = "${rev}";
```
Comment (joachifm) [inline pkgs/misc/emulators/caprice32/default.nix]: Nit: you can do `inherit rev` here, or at least do `rev = rev` ...

File: pkgs/misc/emulators/caprice32/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 = "12yv56blm49qmshpk4mgc802bs51wv2ra87hmcbf2wxma39c45fy";
+  };
+
+  postPatch = "substituteInPlace cap32.cfg --replace /usr/local $out";
+
+  meta = with stdenv.lib; {
+    description = "a complete emulation of CPC464, CPC664 and CPC6128";
```
Comment (joachifm) [inline pkgs/misc/emulators/caprice32/default.nix]: Please capitalize the description

File: pkgs/misc/emulators/caprice32/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    homepage = https://github.com/ColinPitrat/caprice32 ;
+    license = licenses.gpl2;
+    maintainers = [ maintainers.genesis ];
+    platforms = platforms.linux;
+  };
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ libpng SDL freetype zlib ];
+  makeFlags = [ "GIT_HASH=$(src.rev)" "DESTDIR=$(out)" "prefix=/"];
```
Comment (joachifm) [inline pkgs/misc/emulators/caprice32/default.nix]: I wonder if `GIT_HASH=$(src.rev)` will really work as expected ... I suggest using `GIT_HASH=${src.rev}` instead.

**Comment by joachifm** [comment]
@GrahamcOfBorg build caprice32

File: pkgs/misc/emulators/caprice32/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+stdenv.mkDerivation rec {
+
+  repo = "caprice32";
+  version = "unstable-2018-02-10";
+  rev = "53de69543300f81af85df32cbd21bb5c68cab61e";
+  name = "${repo}-${version}";
+
+  src = fetchFromGitHub {
+    inherit rev repo;
```
Comment (volth) [inline pkgs/misc/emulators/caprice32/default.nix]: What was the reason to define ```repo``` and ```rev``` outside and then inherit here?
Why not just here?

Comment (bignaux) [reply pkgs/misc/emulators/caprice32/default.nix]: 'cause i like this, it could be implicit rules.

**Pull request merged**